pkg/controller/siddhiprocess/artifacts.go

```diff
@@ -169,41 +168,53 @@ func (rsp *ReconcileSiddhiProcess) UpdateIngress(
 	}
 
 	currentRules := currentIngress.Spec.Rules
-	newRule := extensionsv1beta1.IngressRule{
-		Host: configs.HostName,
-		IngressRuleValue: extensionsv1beta1.IngressRuleValue{
-			HTTP: &extensionsv1beta1.HTTPIngressRuleValue{
-				Paths: ingressPaths,
-			},
+	ruleValue := extensionsv1beta1.IngressRuleValue{
+		HTTP: &extensionsv1beta1.HTTPIngressRuleValue{
+			Paths: ingressPaths,
 		},
 	}
+	newRule := extensionsv1beta1.IngressRule{
+		Host:             configs.HostName,
+		IngressRuleValue: ruleValue,
+	}
 	ruleExists := false
+	needUpdate := false
 	for _, rule := range currentRules {
-		if reflect.DeepEqual(rule, newRule) {
+		if rule.Host == configs.HostName {
 			ruleExists = true
+			for _, path := range ingressPaths {
+				if !pathContains(rule.HTTP.Paths, path) {
+					needUpdate = true
+					rule.HTTP.Paths = append(rule.HTTP.Paths, path)
+				}
+			}
 		}
 	}
+
 	if !ruleExists {
+		needUpdate = true
```

Comment:
Shall we use the same ruleExists var for updating

Reply:
Here we can come across a scenario where `ruleExists=true` but `needUpdate=false` according to this loop.
```
     for _, rule := range currentRules {
		if rule.Host == configs.HostName {
			ruleExists = true
			for _, path := range ingressPaths {
				if !pathContains(rule.HTTP.Paths, path) {
					needUpdate = true
					rule.HTTP.Paths = append(rule.HTTP.Paths, path)
				}
			}
		}
	}
```
To handle that kind of scenrios we need two different variables.